test/org/sagebionetworks/bridge/upload/IosSchemaValidationHandler2Test.java

```diff
@@ -380,8 +391,9 @@ public void mixedData() throws Exception {
         assertEquals(1, recordBuilder.getSchemaRevision());
 
         JsonNode dataNode = recordBuilder.getData();
-        assertEquals(2, dataNode.size());
+        assertEquals(3, dataNode.size());
         assertEquals("This is a string", dataNode.get("field.json.string").textValue());
+        assertTrue(dataNode.get("optional").isNull());
```

Comment:
Is there a path that tests when it's required?

Other than that, I can merge when you're ready.